cmd/kubeadm/app/preflight/utils_test.go

```diff
@@ -34,6 +34,14 @@ func TestGetKubeletVersion(t *testing.T) {
 	}{
 		{"Kubernetes v1.7.0", "1.7.0", nil, true},
 		{"Kubernetes v1.8.0-alpha.2.1231+afabd012389d53a", "1.8.0-alpha.2.1231+afabd012389d53a", nil, true},
+		{"Kubernetes v1.8.0-alpha.2.1231+afabd012389d53a ", "1.8.0-alpha.2.1231+afabd012389d53a", nil, true},
+		{" Kubernetes v1.8.0-alpha.2.1231+afabd012389d53a", "1.8.0-alpha.2.1231+afabd012389d53a", nil, true},
+		{"Kubernetes  v1.8.0-alpha.2.1231+afabd012389d53a", "", nil, false},
+		{"Kubernetes v1.8.0alpha.2.1231+afabd012389d53a", "", nil, false},
+		{"Kubernetes v1.8.0--alpha.2.1231+afabd012389d53a", "1.8.0--alpha.2.1231+afabd012389d53a", nil, true},
+		{"Kubernetes v1.8.0-+alpha.2.1231+afabd012389d53a", "", nil, false},
+		{"Kubernetes v1.8.0-alpha.2.1231+-afabd012389d53a", "1.8.0-alpha.2.1231+-afabd012389d53a", nil, true},
+		{"Kubernetes v1.8.0-alpha.2.1231++afabd012389d53a", "", nil, false},
```

Comment:
some of these are not supported by semver and formatting of version in the kubelet - e.g. we assume that `v1.8.0-alpha.2.1231++afabd012389d53a` or `Kubernetes  v1.8.0...` will never happen.

Reply:
OK，I will modify it